### PR TITLE
[Translator] move loading catalogue out of Translator.

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | yes |
| Fixed tickets | ~ |
| Tests pass? | yes |
| License | MIT |
#### Before

``` php
$translator = new Translator('fr');
$translator->addLoader('array', new ArrayLoader());
$translator->addResource('array', array('Hello World!' => 'Bonjour'), 'fr');

echo $translator->trans('Hello World!')."\n";
```
#### After

``` php
$resourceCatalogue = new ResourceMessageCatalogueProvider();
$resourceCatalogue->addLoader('array', new ArrayLoader());
$resourceCatalogue->addResource('array', array('Hello World!' => 'Bonjour'), 'fr');
$translator = new Translator('fr', $resourceCatalogue);

echo $translator->trans('Hello World!')."\n";
```

### CODE REVIEW

**Comment by aitboudad** [comment]
@symfony/deciders 
I really want to see this in 2.8 which will allows us to simplify a bunch of things in the translator component.

**Comment by fabpot** [comment]
I'm ok with the decoupling but hadn't had time to review the PR.

File: src/Symfony/Component/Translation/MessageCatalogueProvider/ResourceMessageCatalogueProvider.php [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+use Symfony\Component\Translation\Loader\LoaderInterface;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\Translator;
+
+/**
+ * MessageCatalogueProvider loads catalogue from resources.
+ *
+ * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
+ */
+class ResourceMessageCatalogueProvider implements MessageCatalogueProviderInterface
```
Comment (fabpot) [inline src/Symfony/Component/Translation/MessageCatalogueProvider/ResourceMessageCatalogueProvider.php]: What about just `MessageCatalogueProvider` as this is really the default implementation that almost everyone is going to use, right?

Comment (aitboudad) [inline src/Symfony/Component/Translation/MessageCatalogueProvider/ResourceMessageCatalogueProvider.php]: yes, I'll rename `ContainerAwareResourceMessageCatalogueProvider` too.

Comment (aitboudad) [inline src/Symfony/Component/Translation/MessageCatalogueProvider/ResourceMessageCatalogueProvider.php]: done

**Comment by iltar** [comment]
Even though this PR is rather big and my understanding of the changes minimal, I like what I see here. If possible I would indeed like to see this in 2.8 so it prevents a big BC layer scattered around for a while. I'm really happy to see the framework translator go, it was a pretty confusing setup and might inspire the router to be decoupled like that in the future as well.

**Comment by mpdude** [comment]
Are there major differences to #14671? We've already had a lot of discussion and suggestions there.

**Comment by fabpot** [comment]
@aitboudad Now that 3.0 is out, what about this one?

**Comment by aitboudad** [comment]
see #17563, it only needs to be reviewed

**Pull request closed without merging**